### DISCUSSION TITLE
Duda e ideas...

### DISCUSSION THREAD

**Discussion by AlexandreInking** [gamerevar/proyecto-comunitario-1-gd#55] [Archivados]
Duda: Los juegos seran cada uno desarrollado en distintos engine o se hara en uno de los dos ambos. Sé que al inicio haran por separado en cada uno pero se me ocurre una idea online y no se si se podra implementar en juegos hechos en distintos engines.

Ideas: 
NO ONLINE: Se me ocurre que podría ser un Metroidvania como lei en una discusion anterior en el cual la historia de cada parte sea una forma distinta de ver el mundo, objetivos completamente distintos. Especificamente: un personaje es el heroe de la historia y en una de las misiones se une al villano (sin que el heroe lo sepa) para derrotarlo, mientras que en el otro juego el encuentro con el heroe no es casual sino totalmente planeado y que es parte del plan del villano ganarse la confianza del heroe para luego matarlo por la espalda.

Objetivo del heroe: Derrotar al villano (digamos Emperador Demonio...). Para salvar la humanidad (cuando se entera que los demonios estan siendo manipulados: salvar tambien a los demonios).

Objetivo del villano: Llegar a ser un dios demonio.

Emperador demonio: Un ser que no le importan ni siquiera a los propios demonios, a los cuales ha esclavizado y obliga a atacar inocentes a traves de collares. Según textos antiguos de la tribu demonio se detalla que tras asesinar a 100 mil seres de cada especie es posible llegar a ser un dios.

Mundo de magia (tecnologia no desarrollada, referencia temporal aproximada: Edad de Hierro): Los humanos no pueden usar magia por naturaleza, la magia proviene de los demonios emanan esa energia. El exterminio de los demonios implica el exterminio de la magia.

ONLINE: Juego shooter estrategico de magos (Valorant + Spellbreak) hasta 15 rondas de 2 minutos con habilidades creadas por los mismos jugadores. Un sistema de personalizacion de hechizos que las habilidades de los jugadores puedan ser modificadas por los mismos usuarios.

Ejemplo con el muro de Sage en Valorant: el muro dura 30 segundos pero el usuario podria aumentar el tiempo de duracion sacrificando la resistencia del muro. Haciendo que no se rompa el balance del juego. 

Serían habilidades predeterminadas con variación según el modo de juego del jugador. Al superar misiones se obtengan puntos los cuales puedan ser usados en la tienda para comprar hechizos nuevos.

**Comment by LeandroCuello** [comment]
me gusta la idea ofline suena interesante. Ahora no se si soy fanatico de hacer un juego online, suelen ser muy complicados y mas un shooter fps ya que tendriamos que trabajar en mucho net coding. No se que piensa el resto, tal vez algunos se manejen haciendo cosas online.
respecto a la duda, son 2 juegos uno armado en unity y otro en godot. Los lead disigners tubieron la idea de que los juegos esten conectos de alguna forma (seguramente termine siendo por historia). Ahora todos vamos a hacer brainstorming en el repo de godot por un tema de comidida, una vez que termine la etapa de brainstorming cada grupo va a usar su repo.

**Comment by AlexandreInking** [comment]
**¿Que hace especial al héroe?**
Es un héroe invocado lo cual le otorga un beneficio en stats y velocidad de crecimiento (200% XP) a comparación de otros seres en ese mundo.

**¿Porque acompañar al héroe en su aventura en vez de matarlo mientras está débil?**
1. Necesita matar a 100 mil demonios, va de encubierto (para justificar el asesinato de los que debe matar), como un demonio que se logró escapar del emperador demonio. Tomando el papel de ser el apoyo del héroe y guía en este nuevo mundo.

2. Debe tener una idea del poder del héroe y su desarrollo y superar su poder. Por lo anterior mencionado el emperador demonio debe de mejorar y superar el poder del héroe.

**Comment by jh62** [comment]
Me gusta la idea offline, pero yo personalmente soy más partidario de que ambos juegos estén conectados entre si con vínculos no tan fuertes, por un tema de complejidad de diseño.

**Reply by AlexandreInking** [reply]
Entonces podría ser algo así como que ambos coexisten en el mismo mundo pero no interaccionan hasta el final. Relacionándolo con la idea offline se me ocurre que en un juego se desarrolle la historia de arriba pero desde la perspectiva del héroe y en el otro juego se muestre un demonio que busque vengar a los demonios caídos por la mano del héroe (a diferencia del emperador demonio que le importa una sh*t todos) y que la batalla final contra él sea la apertura de la batalla con el emperador demonio.

**¿Como la apertura de la batalla?**
Cuando uno de ellos (depende del juego) muere, el emperador demonio revela ser quien es y pelea con el jugador (sin tiempo para curarse o cambiar armas), de esta manera, explica que el controlaba a los demonios y manipulaba la codicia de los humanos para generar sus muertes y absorber sus almas, finalmente, muestra su verdadero poder, hasta que es vencido por el protagonista de cada historia. Si el jugador muere en la batalla retrocede al ultimo punto de guardado (preferentemente que esté a las puertas de la batalla contra el demonio vengador/héroe para que no sea tan difícil volver).
